### ISSUE TITLE
[optim] add missing 'maximize' parameter to LBFGS, NAdam and RAdam optimizers

### ISSUE THREAD

**Issue by KebabRonin** [pytorch/pytorch#126642]
### 🚀 The feature, motivation and pitch

Currently, ```LBFGS```, ```NAdam``` and ```RAdam``` optimizers do not have the ```maximize``` parameter in their constructors, which means they cannot be used with some loss functions (like R2Score).

### Alternatives

In the case ```maximize``` can't be implemented due to a constraint of the optimizer, it would be nice to make it clear that ```maximize``` is intentionally not supported. Some possible suggestions:
* Update the documentation to mention the reason maximize isn't supported for the optimizer
* Still support the parameter but throw an error when maximize=True

cc @vincentqb @jbschlosser @albanD @janeyx99 @crcrpar

**Comment by janeyx99** [comment]
Yes, it is known that these do not support the flag, and the documentation for each exclude mentioning maximize for that reason. 

It is totally easily actionable to add maximize for RAdam and NAdam + tests, perhaps easier than adding the docs lol. I may just add this for funsies later this week but if someone wants to do this, go for it!

LBFGS I've gotta check, but off the top of my head it should work as well, but the implementation is not immediately doable.

**Comment by huihoaan** [comment]
@janeyx99 May I help to add this? I have made a pr for `Radam`. If https://github.com/pytorch/pytorch/pull/126765 is OK, I will add another pr for Nadam.

**Comment by janeyx99** [comment]
Reopening to track LBFGS
